torch/testing/_internal/opinfo/core.py

```diff
@@ -252,7 +255,8 @@ def to_numpy(t):
 
     def noncontiguous(self):
         def to_noncontiguous(t):
-            if isinstance(t, torch.Tensor):
+            # currently sparse layouts have continuity invariants that make this impossible
```

Comment:
It would be preferable to skip or x fail in this situation, unfortunately I don't know of any mechanism for conditionally skipping or xfailing based on attributes of the test sample.

Right now we will silently skip generation of non-contiguous sparse inputs but we still exercise the op with non-contiguous dense inputs.

The best intermediate I can think of would be to put code here which generates contiguous sparse and validates them. Expecting validation to fail, should validation of that transformed sparse start passing at some point an exception is raised indicating it needs to be updated. 

That would be a lot of work, as the validation checks are non-trivial.